browser_patches/webkit/embedder/Playwright/mac/AppDelegate.m

```diff
@@ -250,8 +250,18 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification
     if (_noStartupWindow)
         return;
 
-    [self createNewPage:0];
-    _initialURL = nil;
+    // Force creation of the default browser context.
+    [self defaultConfiguration];
+    // Creating the first NSWindow immediately makes it invisible in headless mode,
+    // so we postpone it for 50ms. Experiments show that 10ms is not enough, and 20ms is enough.
+    // We give it 50ms just in case.
+    [NSTimer scheduledTimerWithTimeInterval: 0.05
+                                    repeats: NO
+                                      block:(void *)^(NSTimer* timer)
+    {
+        [self createNewPage:0 withURL:_initialURL ? _initialURL : @"about:blank"];
```

Comment:
Would it make sense to simply create the page on the client side (and always start in _noStartupWindow mode) ?